.github/workflows/c8run-source-build.yaml

```diff
@@ -0,0 +1,201 @@
+name: "C8Run: source build/test"
+
+on:
+  schedule:
+    - cron: "0 23 * * *"
+
+permissions:
+  actions: write
+  attestations: none
+  checks: write
+  contents: read
+  deployments: none
+  id-token: none
+  issues: none
+  discussions: none
+  packages: none
+  pages: none
+  pull-requests: none
+  repository-projects: none
+  security-events: none
+  statuses: write
+
+jobs:
+  linting:
+    name: C8Run linting
+    runs-on: ubuntu-latest
+    timeout-minutes: 5
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-go@v5
+        with:
+        go-version: '>=1.23.1'
+        cache: false  # disabling since not working anyways without a cache-dependency-path specified
+      - name: golangci-lint
+        uses: golangci/golangci-lint-action@ec5d18412c0aeab7936cb16880d708ba2a64e1ae # v6.2.0
+        with:
+          working-directory: ./c8run
+  test_c8run:
+    strategy:
+      matrix:
+        # macos-latest is ARM, mac os 13 will execute on x86 runner.
+        os: [ubuntu-latest, macos-latest, macos-13]
+    name: C8Run Test ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    timeout-minutes: 40
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: actions/setup-node@v4
+        with:
+          node-version: 18
+
+      - name: Setup C8Run
+        uses: ./.github/actions/setup-c8run
+        with:
+          vault-address: ${{ secrets.VAULT_ADDR }}
+          vault-role-id: ${{ secrets.VAULT_ROLE_ID }}
+          vault-secret-id: ${{ secrets.VAULT_SECRET_ID }}
+          os: ${{ matrix.os }}
+					github-token: ${{ github.token }}
+          source-build: true
+
+      - name: Install dependencies
+        run: npm ci
+        working-directory: ./c8run/e2e_tests
+
+      - name: Install Playwright Browsers
+        run: npx playwright install --with-deps
+        working-directory: ./c8run/e2e_tests
+
+      - name: Unit tests
+        run: go test
+        working-directory: ./c8run
+
+      - name: Run Playwright tests
+        run: npx playwright test
+        working-directory: ./c8run/e2e_tests
+
+      - name: Run v2 API endpoint test
+        run: ./api_tests.sh
+        working-directory: ./c8run/e2e_tests
+
+      - uses: actions/upload-artifact@v4
+        with:
+          name: playwright-report-${{ matrix.os }}
+          path: ./c8run/e2e_tests/playwright-report
+          retention-days: 30
+
+      - uses: actions/upload-artifact@v4
+        if: failure()
+        with:
+          name: c8run-logs-${{ matrix.os }}
+          path: ./c8run/log/*.log
+          retention-days: 10
+
+
+					test_c8run_windows:
+						name: C8Run Test Windows
+						runs-on: windows-latest
+						timeout-minutes: 15
+						steps:
+							- uses: actions/checkout@v4
+				
+							- name: Import Secrets
+								id: secrets
+								uses: hashicorp/vault-action@d1720f055e0635fd932a1d2a48f87a666a57906c # v3.0.0
+								with:
+									url: ${{ secrets.VAULT_ADDR }}
```

Comment:
sayy immi, did you by any chance change mid-file on whether you prefer tabs vs spaces?

Reply:
:D omg - just fixed it 🙈